smile/socket_interface.py

```diff
@@ -0,0 +1,156 @@
+#emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+#ex: set sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See the COPYING file distributed along with the smile package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+from state import CallbackState
+from clock import clock
+
+import sys
+import socket
+
+_sockets = {}
+
+
+def init_socket_outlet(uniq_ID, server, port):
+    """Sets up a socket that allows for TCP messaging.
+
+    A *SocketPush* state will use a pre-initialized Socket to send out a
+    marker to the specific server and port. Once created, the socket will be
+    added to a dictionary of sockets (_sockets) where the key is a string joined
+    by an *_*. EX **markersocket_127.0.0.1_1234**.
+
+    Parameters
+    ----------
+    unique_id : string
+        Unique identifier for this socket. Used only to differenciate this
+        socket from other sockets initialized with this funciton.
```

Comment:
funciton -> function
```suggestion
        socket from other sockets initialized with this function.
```